from PyQt5 import QtCore, QtGui, QtWidgets
from translate import Ui_Translate
import time


class Ui_LessonChoice(object):
    def show_main(self, main_window, current_window):
        main_window.show()
        current_window.hide()

    def openTranslate(self, ui_name, lesson_choice):
        self.window = QtWidgets.QMainWindow()
        self.ui = ui_name
        self.ui.setupUi(self.window, lesson_choice)
        lesson_choice.hide()
        self.window.show()

    def setupUi(self, LessonChoice, MainWindow, user):
        self.user = user
        LessonChoice.setObjectName("LessonChoice")
        LessonChoice.resize(500, 450)
        LessonChoice.setStyleSheet("background-color: rgb(53, 132, 228);")
        self.centralwidget = QtWidgets.QWidget(LessonChoice)
        self.centralwidget.setObjectName("centralwidget")
        self.gridLayout = QtWidgets.QGridLayout(self.centralwidget)
        self.gridLayout.setObjectName("gridLayout")
        spacerItem = QtWidgets.QSpacerItem(25, 20, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        self.gridLayout.addItem(spacerItem, 1, 4, 1, 1)
        self.gridLayout_3 = QtWidgets.QGridLayout()
        self.gridLayout_3.setSizeConstraint(QtWidgets.QLayout.SetMinAndMaxSize)
        self.gridLayout_3.setObjectName("gridLayout_3")
        self.allOpenButton = QtWidgets.QPushButton(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.allOpenButton.sizePolicy().hasHeightForWidth())
        self.allOpenButton.setSizePolicy(sizePolicy)
        self.allOpenButton.setMaximumSize(QtCore.QSize(700, 60))
        self.allOpenButton.setStyleSheet("background-color: rgb(129, 61, 156);")
        self.allOpenButton.setObjectName("allOpenButton")
        self.gridLayout_3.addWidget(self.allOpenButton, 4, 0, 1, 1)
        self.pronounceOpenButton = QtWidgets.QPushButton(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.pronounceOpenButton.sizePolicy().hasHeightForWidth())
        self.pronounceOpenButton.setSizePolicy(sizePolicy)
        self.pronounceOpenButton.setMaximumSize(QtCore.QSize(700, 60))
        self.pronounceOpenButton.setStyleSheet("background-color: rgb(192, 97, 203);")
        self.pronounceOpenButton.setObjectName("pronounceOpenButton")
        self.gridLayout_3.addWidget(self.pronounceOpenButton, 2, 0, 1, 1)
        self.translateOpenButton = QtWidgets.QPushButton(self.centralwidget,
                                                         clicked=lambda: self.openTranslate(Ui_Translate(), LessonChoice))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.translateOpenButton.sizePolicy().hasHeightForWidth())
        self.translateOpenButton.setSizePolicy(sizePolicy)
        self.translateOpenButton.setMaximumSize(QtCore.QSize(700, 60))
        self.translateOpenButton.setStyleSheet("background-color: rgb(220, 138, 221);")
        self.translateOpenButton.setObjectName("translateOpenButton")
        self.gridLayout_3.addWidget(self.translateOpenButton, 0, 0, 1, 1)
        self.hearMeOpenButton = QtWidgets.QPushButton(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.MinimumExpanding, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.hearMeOpenButton.sizePolicy().hasHeightForWidth())
        self.hearMeOpenButton.setSizePolicy(sizePolicy)
        self.hearMeOpenButton.setMaximumSize(QtCore.QSize(700, 60))
        self.hearMeOpenButton.setStyleSheet("background-color: rgb(145, 65, 172);")
        self.hearMeOpenButton.setObjectName("hearMeOpenButton")
        self.gridLayout_3.addWidget(self.hearMeOpenButton, 3, 0, 1, 1)
        self.passedButton = QtWidgets.QPushButton(self.centralwidget,
                                                  clicked=lambda: self.show_main(MainWindow, LessonChoice))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Maximum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.passedButton.sizePolicy().hasHeightForWidth())
        self.passedButton.setSizePolicy(sizePolicy)
        self.passedButton.setMaximumSize(QtCore.QSize(700, 60))
        self.passedButton.setStyleSheet("background-color: rgb(97, 53, 131);")
        self.passedButton.setObjectName("passedButton")
        self.gridLayout_3.addWidget(self.passedButton, 5, 0, 1, 1)
        self.gridLayout.addLayout(self.gridLayout_3, 1, 1, 1, 3)
        spacerItem1 = QtWidgets.QSpacerItem(15, 15, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Maximum)
        self.gridLayout.addItem(spacerItem1, 2, 2, 1, 1)
        spacerItem2 = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.gridLayout.addItem(spacerItem2, 1, 0, 1, 1)
        self.gridLayout_2 = QtWidgets.QGridLayout()
        self.gridLayout_2.setContentsMargins(-1, 1, -1, -1)
        self.gridLayout_2.setObjectName("gridLayout_2")
        self.profileOpenButton = QtWidgets.QPushButton(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.profileOpenButton.sizePolicy().hasHeightForWidth())
        self.profileOpenButton.setSizePolicy(sizePolicy)
        self.profileOpenButton.setMinimumSize(QtCore.QSize(80, 30))
        self.profileOpenButton.setMaximumSize(QtCore.QSize(100, 30))
        self.profileOpenButton.setObjectName("profileOpenButton")
        self.gridLayout_2.addWidget(self.profileOpenButton, 1, 2, 1, 1)
        spacerItem3 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.MinimumExpanding, QtWidgets.QSizePolicy.Minimum)
        self.gridLayout_2.addItem(spacerItem3, 1, 1, 1, 1)
        spacerItem4 = QtWidgets.QSpacerItem(20, 58, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        self.gridLayout_2.addItem(spacerItem4, 1, 0, 1, 1)
        self.gridLayout.addLayout(self.gridLayout_2, 0, 1, 1, 4)
        self.verticalLayout = QtWidgets.QVBoxLayout()
        self.verticalLayout.setObjectName("verticalLayout")
        spacerItem5 = QtWidgets.QSpacerItem(25, 9, QtWidgets.QSizePolicy.Maximum, QtWidgets.QSizePolicy.Minimum)
        self.verticalLayout.addItem(spacerItem5)
        self.gridLayout.addLayout(self.verticalLayout, 0, 0, 1, 1)
        LessonChoice.setCentralWidget(self.centralwidget)

        self.retranslateUi(LessonChoice)
        QtCore.QMetaObject.connectSlotsByName(LessonChoice)

    def retranslateUi(self, LessonChoice):
        _translate = QtCore.QCoreApplication.translate
        LessonChoice.setWindowTitle(_translate("LessonChoice", "Lessons"))
        self.allOpenButton.setText(_translate("LessonChoice", "All in one"))
        self.pronounceOpenButton.setText(_translate("LessonChoice", "Pronounce"))
        self.translateOpenButton.setText(_translate("LessonChoice", "Translate "))
        self.hearMeOpenButton.setText(_translate("LessonChoice", "Do you hear me?"))
        self.passedButton.setText(_translate("LessonChoice", "Back"))
        self.profileOpenButton.setText(_translate("LessonChoice", "Profile"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    LessonChoice = QtWidgets.QMainWindow()
    ui = Ui_LessonChoice()
    ui.setupUi(LessonChoice)
    LessonChoice.show()
    sys.exit(app.exec_())
